client/blocks/login/login-header.tsx

```diff
@@ -0,0 +1,597 @@
+					</span>
+				) }
+			</p>
+		);
+		if ( isBlazePro ) {
+			postHeader = (
+				<p className="login__header-subtitle login__lostpassword-subtitle">
+					{ translate(
+						'It happens to the best of us. Enter the email address associated with your Blaze Pro account and we’ll send you a link to reset your password.'
+					) }
+				</p>
+			);
+		}
+	} else if ( oauth2Client ) {
+		if ( isWCCOM ) {
+			if ( wccomFrom === 'cart' ) {
+				preHeader = <WooCommerceConnectCartHeader />;
+				postHeader = (
+					<p className="login__header-subtitle">
+						{ translate(
+							'Log in to WooCommerce.com with your WordPress.com account to connect your store and manage your extensions'
+						) }
+					</p>
+				);
+			} else if ( twoFactorEnabled ) {
+				header = <h3>{ headerText }</h3>;
+			} else if ( currentQuery.lostpassword_flow ) {
+				header = <h3>{ headerText }</h3>;
+				postHeader = (
+					<p className="login__header-subtitle">
+						{ translate(
+							"Your password reset confirmation is on its way to your email address – please check your junk folder if it's not in your inbox! Once you've reset your password, head back to this page to log in to your account."
+						) }
+					</p>
+				);
+			} else if ( showContinueAsUser() ) {
+				header = <h3>{ headerText }</h3>;
+				postHeader = (
+					<p className="login__header-subtitle">
+						{ wccomFrom === 'nux'
+							? translate( 'First, select the account you’d like to use.' )
+							: translate( 'Select the account you’d like to use.' ) }
+					</p>
+				);
+			} else {
+				header = <h3>{ headerText }</h3>;
+				const poweredByWpCom = (
+					<>
+						{ translate( 'Log in with your WordPress.com account.' ) }{ ' ' }
+						<br className="hide-on-desktop" />
+					</>
+				);
+
+				postHeader = (
+					<p className="login__header-subtitle">
+						{ poweredByWpCom }
+						{ translate( "Don't have an account? {{signupLink}}Sign up{{/signupLink}}", {
+							components: {
+								signupLink,
+								br: <br />,
+							},
+						} ) }
+					</p>
+				);
+			}
+		}
+
+		if ( isJetpackCloudOAuth2Client( oauth2Client ) ) {
+			preHeader = (
+				<div>
+					<JetpackPlusWpComLogo className="login__jetpack-plus-wpcom-logo" size={ 24 } />
+				</div>
+			);
+		}
+
+		if ( isA4AOAuth2Client( oauth2Client ) ) {
+			preHeader = (
+				<div>
+					<A4APlusWpComLogo className="login__a4a-plus-wpcom-logo" size={ 32 } />
+				</div>
+			);
+		}
+
+		if ( isPartnerPortalOAuth2Client( oauth2Client ) ) {
+			if ( document.location.search?.includes( 'wpcloud' ) ) {
+				preHeader = (
+					<div>
+						<WPCloudLogo className="login__wpcloud-logo" size={ 256 } />
+					</div>
+				);
+			} else if ( document.location.search?.includes( 'jetpack' ) ) {
+				preHeader = (
+					<div>
+						<JetpackPlusWpComLogo className="login__jetpack-plus-wpcom-logo" size={ 24 } />
+					</div>
+				);
+			}
+		}
+
+		if ( isJetpackCloudOAuth2Client( oauth2Client ) || isA4AOAuth2Client( oauth2Client ) ) {
```

Comment:
I guess it was like this before and we don't want to refactor too much, but the number of permutations/conditions here is pretty mind boggling!

How is it possible to test all these 😱 

I guess the answer is that it isn't. 😄

Reply:
😂 😭